openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BasicAuthenticator.java

```diff
@@ -0,0 +1,477 @@
+    // passwords validity
+    if (!request.getPassword().equals(request.getConfirmPassword())) {
+      throw new RuntimeException("Password and Confirm Password should match");
+    }
+    PasswordUtil.validatePassword(request.getPassword());
+
+    String newHashedPwd = BCrypt.withDefaults().hashToString(12, request.getPassword().toCharArray());
+    BasicAuthMechanism newAuthForUser = new BasicAuthMechanism().withPassword(newHashedPwd);
+
+    storedUser.setAuthenticationMechanism(new AuthenticationMechanism().withAuthType(BASIC).withConfig(newAuthForUser));
+
+    userRepository.createOrUpdate(uriInfo, storedUser);
+
+    // delete the user's all password reset token as well , since already updated
+    tokenRepository.deleteTokenByUserAndType(storedUser.getId().toString(), PASSWORD_RESET.toString());
+
+    // Update user about Password Change
+    try {
+      EmailUtil.getInstance().sendAccountStatus(storedUser, "Update Password", "Change Successful");
+    } catch (Exception ex) {
+      LOG.error("Error in sending Password Change Mail to User. Reason : " + ex.getMessage());
+      throw new CustomExceptionMessage(424, EMAIL_SENDING_ISSUE);
+    }
+    loginAttemptCache.recordSuccessfulLogin(request.getUsername());
+  }
+
+  @Override
+  public void changeUserPwdWithOldPwd(UriInfo uriInfo, String userName, ChangePasswordRequest request)
+      throws IOException {
+    // passwords validity
+    if (!request.getNewPassword().equals(request.getConfirmPassword())) {
+      throw new RuntimeException("Password and Confirm Password should match");
+    }
+    PasswordUtil.validatePassword(request.getNewPassword());
+    List<String> fields = userRepository.getAllowedFieldsCopy();
+    fields.add(USER_PROTECTED_FIELDS);
+
+    // Fetch user
+    User storedUser =
+        userRepository.getByName(uriInfo, userName, new EntityUtil.Fields(fields, String.join(",", fields)));
+    BasicAuthMechanism storedBasicAuthMechanism =
+        JsonUtils.convertValue(storedUser.getAuthenticationMechanism().getConfig(), BasicAuthMechanism.class);
+    if (request.getRequestType() == SELF) {
+      String storedHashPassword = storedBasicAuthMechanism.getPassword();
+      if (BCrypt.verifyer().verify(request.getOldPassword().toCharArray(), storedHashPassword).verified) {
+        String newHashedPassword = BCrypt.withDefaults().hashToString(12, request.getNewPassword().toCharArray());
+        storedBasicAuthMechanism.setPassword(newHashedPassword);
+        storedUser.getAuthenticationMechanism().setConfig(storedBasicAuthMechanism);
+        userRepository.createOrUpdate(uriInfo, storedUser);
+        // remove login/details from cache
+        loginAttemptCache.recordSuccessfulLogin(userName);
+      } else {
+        throw new CustomExceptionMessage(UNAUTHORIZED, "Old Password is not correct");
+      }
+    } else {
+      String newHashedPassword = BCrypt.withDefaults().hashToString(12, request.getNewPassword().toCharArray());
```

Comment:
Aren't these if else's doing the something if so can we move to a method on changing password and check if the user is the one who is changing or admin